### ISSUE TITLE
Which version of TensorFlow, TensorFlow-GPU and Keras would be best for the CUDA 10.0 for training Mask RCNN?

### ISSUE THREAD

**Issue by subbulakshmisubha** [matterport/Mask_RCNN#1858]
Which version of TensorFlow, TensorFlow-GPU and Keras would be best for the CUDA 10.0 for training Mask RCNN?. I tried multiple combinations looking at the issues but kept getting issues.

**Comment by sebimarkgraf** [comment]
Try the versions mentioned in the README:
> Python 3.4, TensorFlow 1.3, Keras 2.0.8

Tried with Python 3.6 but shouldn't make much of a difference.

TF2 is currently not supported and several PRs are working on it.

#1835  #1815

**Comment by MGarvinNYC** [comment]
@Sintifo:  Can you elaborate on your comment about ongoing work to port to TF2?  Is there any way to monitor the project?

**Comment by sebimarkgraf** [comment]
Disclaimer: I am not any official mantainer of this project

There are multiple PRs open #1847 #1842 #1817 that fix the basic movement and renaming of functions. But as I mentioned #1835 in my earliert comment, tf2 does not call `compute_output_shape` and therefore the outputs are not in the correct format.

Therefore to use tensorflow 2 with Mask_RCNN the issue https://github.com/tensorflow/tensorflow/issues/33785 needs to be fixed first. 

@kiflowb777 seems to be actively working on it. So maybe, he can provide you more information.

**Comment by kiflowb777** [comment]
On page https://www.tensorflow.org/community/roadmap in point **Roadmap/Reference models**, It says that the mask-RCNN will be supported by TensorFlow 2.0.

**Comment by kiflowb777** [comment]
Before I started transferring my project to the new TensorFlow 2.0 I used versions:
tf 1.14, keras 2.2.4 (not tf.keras which also don't work in previous tf 1.x versions with mask-RCNN) and python 3.6.1

**Comment by kiflowb777** [comment]
I can see that something is being fixed in this matter: https://github.com/tensorflow/tensorflow/pull/34248
Maybe in the next version of the TF it will be easy to move the Mask-RCNN to tf.keras.

**Comment by mangalbhaskar** [comment]
I have shared my experience on this here [#1808](https://github.com/matterport/Mask_RCNN/issues/1808), [issuecomment-544436458](https://github.com/matterport/Mask_RCNN/issues/1808#issuecomment-544436458) and [issuecomment-557645482](https://github.com/matterport/Mask_RCNN/issues/1808#issuecomment-557645482)

**Comment by thomasjv799** [comment]
Use Tensorflow-gpu 1.15.0 and Keras  2.1.6

**Comment by burhr2** [comment]
Hi! kindly see this issue https://github.com/matterport/Mask_RCNN/issues/2312 there is a pointer to the same implementation that supports TensorFlow 2=> (with GPU). If this solves your problem kindly close the issue so as others can navigate to other issues easier

**Comment by Oscarjia** [comment]
Actually tensorflow have no version of 1.3. Am i right?

**Comment by haseeb33** [comment]
@Oscarjia r1.3 is one of the early releases of TensorFlow. 
See all the ```tensorflow``` versions.
https://www.tensorflow.org/versions
